File: packages/android/src/main/java/com/kozmos/components/AdaptiveMapShell/AdaptiveMapShell.figma.kt
package com.kozmos.components.adaptivemapshell

import androidx.compose.runtime.Composable
import com.figma.code.connect.Figma
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=1351-7985")
class KozmosAdaptiveMapShellConnect {
    @FigmaProperty(FigmaType.Text, "Panel Label Text")
    val panelLabel: String = "Panel slot"

    @FigmaProperty(FigmaType.Enum, "PanelPlacement")
    val panelPlacement: KozmosMapPanelPlacement = Figma.mapping(
        "Start" to KozmosMapPanelPlacement.Start,
        "End" to KozmosMapPanelPlacement.End
    )

    @Composable
    fun ComponentExample() {
        KozmosAdaptiveMapShell(
            map = {},
            mapLabel = "Venue map",
            panelLabel = panelLabel,
            panelPlacement = panelPlacement
        )
    }
}

File: packages/ios/Sources/Components/AdaptiveMapShell/AdaptiveMapShell.figma.swift
import SwiftUI
import Figma

struct KozmosAdaptiveMapShellConnect: FigmaConnect {
    let component = KozmosAdaptiveMapShell<EmptyView, EmptyView, EmptyView, EmptyView, EmptyView>.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=1351-7985"

    @FigmaString("Panel Label Text")
    var panelLabel: String = "Panel slot"

    @FigmaEnum(
        "PanelPlacement",
        mapping: [
            "Start": KozmosAdaptiveMapShell<EmptyView, EmptyView, EmptyView, EmptyView, EmptyView>.PanelPlacement.start,
            "End": KozmosAdaptiveMapShell<EmptyView, EmptyView, EmptyView, EmptyView, EmptyView>.PanelPlacement.end
        ]
    )
    var panelPlacement: KozmosAdaptiveMapShell<EmptyView, EmptyView, EmptyView, EmptyView, EmptyView>.PanelPlacement = .end

    var body: some View {
        KozmosAdaptiveMapShell(
            mapLabel: "Venue map",
            panelLabel: self.panelLabel,
            panelPlacement: self.panelPlacement,
            map: { EmptyView() },
            mapStatusContent: { EmptyView() },
            controls: { EmptyView() },
            topBar: { EmptyView() },
            panel: { EmptyView() }
        )
    }
}
